torch_geometric/transforms/pad.py

```diff
@@ -0,0 +1,548 @@
+
+
+@functional_transform('pad')
+class Pad(BaseTransform):
+    r"""Applies padding to enforce consistent tensor shapes
+    (functional name: :obj:`pad`).
+
+    This transform will pad node and edge features up to a maximum allowed
+    size in the node or edge feature dimension. By default :obj:`0.0` is used
+    as the padding value and can be configured by setting :obj:`node_pad_value`
+    and :obj:`edge_pad_value`.
+
+    Note that in case of applying Pad to
+    :class:`~HeteroData` input, the :obj:`node_pad_value`
+    or :obj:`edge_pad_value` can be a single float or a Dict[<node type>,
+    <pad value>]. In the latter case, if the user does not include a particular
+    node type in the dict, this node type's attributes will not get padded. In
+    contrast, if the user does not include a particular edge type in the dict,
+    for which the destination node padding was specified, the default edge
+    padding value is used.
+
+    Note that in order to allow for at least one padding node for any padding
+    edge, below conditions must be met:
+
+    * if :obj:`max_num_nodes` is a single value, it must be greater than the
+      maximum number of nodes of any graph in the dataset;
+    * if :obj:`max_num_nodes` is a dictionary, value for every node type must
+      be greater than the maximum number of this type nodes of any graph in the
+      dataset.
+
+    Args:
+        max_num_nodes (int or Dict[str, int]): The maximum number of nodes.
+            In heterogeneous graphs, may also take in a dictionary denoting the
+            maximum number of nodes for specific node types.
+        max_num_edges (int or Dict[Tuple[str, str, str], int], optional):
+            The maximum number of edges.
+            In heterogeneous graphs, may also take in a dictionary denoting the
+            maximum number of edges for specific edge types.
+        node_pad_value (float or Dict[str, float], optional): The fill value to
+            use for node features (default: :obj:`0.0`).
+            In heterogeneous graphs, may also take in a dictionary denoting the
+            fill value for specific node types.
+        edge_pad_value (float or Dict[Tuple[str, str, str], float], optional):
+            The fill value to use for edge features (default: :obj:`0.0`).
+            In heterogeneous graphs, may also take in a dictionary denoting the
+            fill value for specific edge types.
+            Note that in case of :obj:`edge_index` feature the tensors are
+            padded with the first padded node index (which represents a set of
+            self loops on the padded node).
+        train_mask_pad_value (bool, optional): The fill value to use for
+            :obj:`train_mask` (default: :obj:`False`).
+        test_mask_pad_value (bool, optional): The fill value to use for
+            :obj:`test_mask` (default: :obj:`False`).
+        include_keys (List[str] or Tuple[str], optional): Keys to include from
```

Comment:
Shall we just make this `mask_pad_value`?

Reply:
@hatemhelal, do you know if we have any use cases for having separate masks?

Reply:
Not sure I have a use case for separate masks at the moment.